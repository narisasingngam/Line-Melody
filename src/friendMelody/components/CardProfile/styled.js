import styled from "styled-components";

export const Picture = styled.img`
    width: 150px;
    display: flex;
    justify-content: center;
    border-radius: 8px;

    @media only screen and (max-width: 600px)  {
        width: 50px;
  }
`