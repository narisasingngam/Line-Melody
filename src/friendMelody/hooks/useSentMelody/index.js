import { useCallback, useState } from "react";
import axios from "axios";

import { BASE_API } from "./../../constants";

export default function useSentMelody(userId) {
  const [showInput, setInput] = useState(false);
  const [inputMelody, setInputMelody] = useState();
  const [invalidInput, setInvalid] = useState(false)

  const handleClickButton = useCallback(() => {
    setInput(true);
  }, [setInput]);

  const handleInputChange = useCallback(
    (e) => {
      setInputMelody(e.target.value);
    },
    [setInputMelody]
  );

  const handleSentMelody = useCallback(async () => {
    let sucess = "";
    setInvalid(false)

    await axios
      .get(`${BASE_API}api/ownership?userId=${userId}&melodyId=${inputMelody}`)
      .then((respones) => {
        sucess = respones.status;
      });

    if (sucess === "available") {
      await axios
        .post(`${BASE_API}api/purchase`, {
          userId: userId,
          melodyId: inputMelody,
        })
        .then((response) => {
          console.log(response);
        });
    }else {
        setInvalid(true)
    }
  }, [userId, inputMelody, setInvalid]);

  return {
    handleSentMelody,
    handleClickButton,
    handleInputChange,
    showInput,
    inputMelody,
    invalidInput
  };
}
