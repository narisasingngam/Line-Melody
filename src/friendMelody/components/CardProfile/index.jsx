import React from "react";
import { Picture } from "./styled";
import useSentMelody from "./../../hooks/useSentMelody";

const CardProfile = ({ image, name, userId }) => {
  const {
    handleSentMelody,
    handleClickButton,
    handleInputChange,
    showInput,
    invalidInput,
  } = useSentMelody(userId);

  return (
    <div>
      <Picture src={image} />

      <h4>{`Name your friend: ${name}`}</h4>
      <button onClick={handleClickButton}>Input Melody</button>
      {showInput && (
        <div>
          <input onChange={handleInputChange} />
          <button onClick={handleSentMelody}>Sent Melody to your friend</button>
        </div>
      )}
      <div>
          {
              invalidInput && <div>Invalid melody: You sent on ower melody</div>
          }
      </div>
    </div>
  );
};

export default CardProfile;
