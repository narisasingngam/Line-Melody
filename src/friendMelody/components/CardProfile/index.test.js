import React from "react";
import {render} from '@testing-library/react'
import CardProfile from ".";
import useSentMelody from './../../hooks/useSentMelody'
import 'jest-styled-components'

jest.mock("./../../hooks/useSentMelody")

describe("CardProfile", () => {
    const handleSentMelodySpy = jest.fn()
    const handleClickButtonSpy = jest.fn()
    const handleInputChangeSpy = jest.fn()


    beforeEach(() => {
        useSentMelody.mockReturnValue({
            handleSentMelody: handleSentMelodySpy,
            handleClickButton: handleClickButtonSpy,
            handleInputChange: handleInputChangeSpy,
            showInput: 1,
            invalidInput: true,
        })
    })

  it("renders CardProfile correctly", () => {
    const {container} = render(<CardProfile image={'url'} name={"name"} userId={3}/>)

    expect(useSentMelody).toBeCalledWith(3)

    expect(container.firstChild).toMatchSnapshot()
  });
});