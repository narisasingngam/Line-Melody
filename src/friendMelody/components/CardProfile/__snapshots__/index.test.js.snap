// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardProfile renders CardProfile correctly 1`] = `
.c0 {
  width: 150px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 8px;
}

@media only screen and (max-width:600px) {
  .c0 {
    width: 50px;
  }
}

<div>
  <img
    class="c0"
    src="url"
  />
  <h4>
    Name your friend: name
  </h4>
  <button>
    Input Melody
  </button>
  <div>
    <input />
    <button>
      Sent Melody to your friend
    </button>
  </div>
  <div>
    <div>
      Invalid melody: You sent on ower melody
    </div>
  </div>
</div>
`;
